docs/VLSI/Basic-Flow.rst

```diff
@@ -0,0 +1,217 @@
+
+Hence, if we want to monolitically place-and-route the entire SoC with the default tech plug-in parameters for power-straps and corners, the relevant command would be
+.. code-block:: shell
+
+    make par CONFIG=<chipyard_config_name> tech_name=<tech_name> INPUT_CONFS="example-design.yml example-tools.yml example-tech.yml"
+
+In a more typical scenario of working on a single module, for example the Gemmini accelerator within the GemminiRocketConfig Chipyard SoC configuration,
+
+.. code-block:: shell
+  vlsi.inputs.placement_constraints:
+    - path: "Gemmini"
+      type: toplevel
+      x: 0
+      y: 0
+      width: 300
+      height: 300
+      margins:
+        left: 0
+        right: 0
+        top: 0
+        bottom: 0
+
+The relevant ``make`` command would then be
+.. code-block:: shell
+
+    make par CONFIG=GemminiRocketConfig VLSI_TOP=Gemmini tech_name=tsmintel3 INPUT_CONFS="example-design.yml example-tools.yml example-tech.yml"
+
+
+Place-and-route generally requires more fine-grained input specifications regarding power nets, clock nets, pin assignments and floorplanning. While the template configuration files provide defaults for automatic tool defaults, these will usually result in very bad QoR, and therefore it is recommended to specify better-informed floorplans, pin assignments and power nets. For more information about cutomizing theses parameters, please refer to the :ref:`Customizing Your VLSI Flow in Hammer` sections or to the Hammer documentation. 
+Additionally, some Hammer process technology plugins do not provide sufficient default values for requires settings such as power nets and pin assignments (for example, ASAP7). In those cases, these contraint will need to be specified manually in the top-level configuration yml files, as is the case in the ``example-asap7.yml`` configuration file.
+
+Place-and-route tools are very sensitive to process technologes (significantly more sensitive than synthesis tools), and different process technologies may work only on specific tool versions. It is recommended to check what is the appropriate tool version for the specific process technology you are working with.
+
+
+.. Note:: If you edit the yml configuration files in between synthesis and place-and-route, the `make par` command will automatically re-run synthesis. If you would like to avoid that and are confident that your configuration file changes do not affect synthesis results, you may use the `make redo-par` instead.
```

Comment:
I think this note should point to documentation on the hammer chipyard makefile integration (which I don't think exists), but would describe the ability to set some fancy makefile variables. Setting `INPUT_CONFS` makes all the yamls requirements for all actions, but the fancy variables let you set which yamls are necessary for which actions useful for things like changing floorplans or the like that only affect PnR.

Reply:
But that page doesn't exist

Reply:
It seems I thought right. I can probably draft something with or for you offline.

Reply:
Don't you need to add stuff to `HAMMER_EXTRA_ARGS` when running `redo-par` to use your updated ymls? Otherwise Hammer will just read cached results from `par-input.json` without regenerating them.